pages/storage.tsx

```diff
@@ -0,0 +1,270 @@
+      <Button onClick={submit}>Upload</Button>
+      <FileListing />
+    </div>
+  );
+};
+
+const StyledForm = styled(Form)`
+  width: 40em;
+
+  & > * {
+    margin: 0.5em;
+  }
+`;
+
+const FileEntryText = styled(Text)``;
+
+const FileEntryRoot = styled(Button).attrs((props) => ({
+  type: "text",
+  ...props,
+}))`
+  display: flex;
+  justify-content: space-between;
+  height: 2em;
+  flex-direction: row;
+  width: 100%;
+
+  & > * {
+    place-self: center;
+  }
+
+  ${hoverscale(1.025)}
+`;
+
+const DeleteFileButton = styled(Button)`
+  ${hoverscale()}
+`;
+
+type ModalType = {
+  enabled: boolean;
+  title: string;
+  contents: string;
+};
+
+const useModal = (filename: string = "") => {
+  const [modal, setModal] = useState({
+    enabled: false,
+    title: filename ?? "",
+    contents: "",
+  });
+
+  useEffect(() => {
+    if (!modal.title || modal.title === "") {
+      setModal((m) => ({ ...m, contents: "" }));
+      return;
+    }
+    Storage.get(modal.title, { level: "private", download: true })
+      // .then((data) => "")
+      .then((data) => data?.Body?.text?.())
+      .then((contents) => setModal((m) => ({ ...m, contents })))
+      .catch((err) => setModal((m) => ({ ...m, title: "", contents: err })));
+  }, [modal, setModal]);
+
+  return useMemo(
+    () => ({
+      modal,
+      setModal,
+      showModal: () => setModal((m) => ({ ...m, enabled: true })),
+      handleOk: () => setModal((m) => ({ ...m, enabled: false, title: "" })),
+      handleCancel: () =>
+        setModal((m) => ({ ...m, enabled: false, title: "" })),
+    }),
+    [modal, setModal]
+  );
+};
+
+const FileEntry = ({
+  filename,
+  setModal,
+}: {
+  filename: string;
+  setModal?: Dispatch<SetStateAction<ModalType>>;
+}) => {
+  const remove = useCallback(
+    (e: MouseEvent<HTMLElement>) => {
+      e.preventDefault();
+      e.stopPropagation();
+      Storage.remove(filename, { level: "private" });
+    },
+    [filename]
+  );
+
+  const show = useCallback(
+    () => setModal((m) => ({ ...m, enabled: true, title: filename })),
+    [filename, setModal]
+  );
+
+  return (
+    <FileEntryRoot onClick={show}>
+      <FileEntryText>{filename}</FileEntryText>
+      <DeleteFileButton type="text" onClick={remove}>
+        <CloseOutlined />
+      </DeleteFileButton>
+    </FileEntryRoot>
+  );
+};
+
+const FileListingRoot = styled.div`
+  border-radius: 5px;
+  border: 1px solid lightgray;
+  padding: 0.5em;
+
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const FileListing = ({ ...props }) => {
+  const [files, setFiles] = useState<string[]>([]);
+  const { modal, setModal, handleOk, handleCancel } = useModal();
+
+  useEffect(() => {
+    const interval = setInterval(
+      () =>
+        Storage.list("", { level: "private" })
+          .then((result) => {
+            setFiles(result.map((r) => r.key));
+          })
+          .catch((err) => {
```

Comment:
Remove the unused function parameter "err" or rename it to "_err" to make intention explicit. Code smell.